utils.js

```diff
@@ -0,0 +1,66 @@
+function shuffle(array) {
+  array = array.slice(); // Shallow clone array
+  let currentIndex = array.length, temporaryValue, randomIndex;
+
+  // While there remain elements to shuffle...
+  while (currentIndex !== 0) {
+
+    // Pick a remaining element...
+    randomIndex = Math.floor(Math.random() * currentIndex);
+    currentIndex -= 1;
+
+    // And swap it with the current element.
+    temporaryValue = array[currentIndex];
+    array[currentIndex] = array[randomIndex];
+    array[randomIndex] = temporaryValue;
+  }
+
+  return array;
+}
+
+module.exports.shuffle = shuffle;
+
+function removeRanges(array, ranges) {
+  array = array.slice(); // Shallow clone array
+
+  ranges.sort(
+    (a, b) => {
+      const startDiff = a.start - b.start;
+
+      if (startDiff === 0) {
+        return a.end - b.end;
+      }
+
+      return startDiff;
+    }
+  );
+
+  // it's possible that ranges could overlap; make sure to merge them into one range
```

Comment:
ohhhhh, I thought this was a warning for the coder! You did this up right, huh? That's pretty freakin' sweet.